src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/VerifyThingAttachedToCertificate.java

```diff
@@ -33,27 +35,29 @@
      * @param iotAuthClient IoT auth client
      * @param thingRegistry Thing Registry
      * @param networkState  Network state
+     * @param clock a clock instance
      */
     @Inject
     public VerifyThingAttachedToCertificate(IotAuthClient iotAuthClient, ThingRegistry thingRegistry,
-                                            NetworkStateProvider networkState) {
+                                            NetworkStateProvider networkState, Clock clock) {
         this.iotAuthClient = iotAuthClient;
         this.thingRegistry = thingRegistry;
         this.networkState = networkState;
+        this.clock = clock;
     }
 
     private boolean verifyLocally(Thing thing, String certificateId) {
-        logger.atInfo().kv("thing", thing.getThingName()).kv("certificate", certificateId)
+        logger.atDebug().kv("thing", thing.getThingName()).kv("certificate", certificateId)
                 .log("Network down, verifying thing attached to certificate locally");
-        return thing.isCertificateAttached(certificateId);
+        return thing.isCertificateAttached(certificateId, clock);
```

Comment:
I know this makes testing easy, just something to discuss:
I feel there is a lot of flexibility with time here, with every usecase entity passing their own instance of clock. It makes sense for registries to have a clock, but should usecases (or anything using registries) track their own time (clock)?